overlord/devicestate/devicestate.go

```diff
@@ -345,6 +345,20 @@ func remodelTasks(st *state.State, current, new *asserts.Model, deviceCtx snapst
 		}
 		tss = append(tss, ts)
 	}
+	// add new kernel
+	if current.Kernel() != new.Kernel() {
+		// TODO: we need to support corner cases here like:
+		//  0. start with "old-kernel"
+		//  1. remodel to "new-kernel"
+		//  2. remodel back to "old-kernel"
+		// In step (2) we will get a "already-installed" error
+		// here right now (workaround: remove "old-kernel")
```

Comment:
if we keep multiple kernels around we need to fix or kill snapstate.KernelInfo, not needing it would be best. That should probably be done in a prereq.

Reply:
this was addressed in #7115